libcontainer/container_linux.go

```diff
@@ -371,6 +371,14 @@ func (c *linuxContainer) start(process *Process) error {
 }
 
 func (c *linuxContainer) Signal(s os.Signal, all bool) error {
+	status, err := c.currentStatus()
+	if err != nil {
+		return err
+	}
+	// to avoid a PID reuse attack
+	if status == Stopped {
```

Comment:
This check should be `status == Running || status == Created`. According to the runtime-spec:

> Attempting to send a signal to a container that is neither created nor running MUST have no effect on the container and MUST generate an error.

Aside from this, LGTM.